### DISCUSSION TITLE
What are the .p files generated in example 2 and how do I read them?

### DISCUSSION THREAD

**Discussion by mlidGWT** [NLRWindSystems/WEIS#267] [Q&A] [answered]
Hello, 

I am running example 02 using the `*_loads` input files, and I'm wondering what the .p files are that are generated within `output/iteration_0/` including the `timeseries/` subfolder? What is the recommended way to open/view/parse them? They seem to be a binary file when opened in a text editor.

Thanks in advance!

**Comment by dzalkind** [comment, marked as answer]
Hi Meg,

The `.p` pickle files are python binaries saved in WEIS to keep track of the time series and summary stats for each design iteration.

Some jupyter notebooks for parsing them can be found [here](https://github.com/WISDEM/WEIS/tree/main/examples/16_postprocessing) in `rev_DLCs_WEIS.ipynb`.  

You can disable their generation by setting `save_timeseries` and `save_iterations` to false. https://github.com/WISDEM/WEIS/blob/759962e8b26cd98729fbb6f4c20b1bd08e81701c/docs/inputs/weis_modeling_schema.rst#L56

I hope this helps.

Best, Dan
